hackathon_site/hardware/views.py

```diff
@@ -43,6 +44,19 @@ def get(self, request, *args, **kwargs):
         return self.retrieve(request, *args, **kwargs)
 
 
+class IncidentListView(mixins.ListModelMixin, generics.GenericAPIView):
+    queryset = Incident.objects.all().select_related(
+        "order_item", "order_item__order__team"
+    )
+    serializer_class = IncidentSerializer
+
+    filter_backends = (filters.DjangoFilterBackend, SearchFilter)
```

Comment:
`SearchFilter` is unused here, since you also have to specify what fields can be searched on.

What do you think about these fields?

```python
    search_fields = (
        "state",
        "order_item__order__team__team_code",
        "order_item__hardware__name",
        "order_item__hardware__manufacturer"
    )
```

With those, update the queryset to be:

```python
    queryset = Incident.objects.all().select_related(
        "order_item", "order_item__order__team", "order_item__hardware"
    )
```

Then just one simple test case where you do

```python
url = self._build_filter_url(search="name")
```

and this will be good to go!